Tests/SentryTests/SentryScopeSwiftTests.swift

```diff
@@ -214,4 +214,70 @@ class SentryScopeSwiftTests: XCTestCase {
         
         XCTAssertEqual(1, attachments.count)
     }
+    
+    // Altough we only run this test above the below specified versions, we exped the
+    // implementation to be thread safe
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testModifyingScopeFromMultipleThreads() {
+        let queue = DispatchQueue(label: "SentryScopeTests", qos: .utility, attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+        
+        let scope = fixture.scope
+        
+        for _ in Array(0...100) {
+            group.enter()
+            queue.async {
+                
+                let key = "key"
+                
+                _ = Scope(scope: scope)
```

Comment:
I'd try instead of 100, 100_000.

Ideally we'd run this for a few hours (not in CI). I suggest removing some of the locks and having it crash, otherwise this isn't doing what you think

Reply:
I tried 100_000 and found one issue in the test. I had to remove `XCTAssertFalse(scope.attachments.isEmpty)`, making sense because below we clear the scope. I already removed each lock several times to make sure it crashes. So I think we are good.